### DISCUSSION TITLE
Using kci_build

### DISCUSSION THREAD

**Discussion by ark-sys** [kernelci/kernelci-core#619] [Q&A]
Hi, I'm an IT student and I'm currently using kernelci in order to build and test some kernel versions.
I would like to have some advice on how to use the command kci_build with the build_kernel option. What i want to do is to download a tarball from "https://mirrors.edge.kernel.org/pub/linux/kernel/v5.x/", unpack it and use it as kdir with a custom .config in order to build a kernel.

So, how can i pass a custom .config to kci_build?
Can i use a specific kernel version from a tarball without needing to update local mirrors or defining config-trees?
Thanks

**Comment by mgalka** [comment]
Hi,
There has been quite a few changes introduced to `kci_build` recently. You may want to have a look
into the docs which are available at the moment on our staging instance https://static.staging.kernelci.org/docs/core/kci_build/

If you want to use your own `.config` it should be sufficient to skip the `kci_build make_config` step.
